pandas/core/strings.py

```diff
@@ -999,23 +999,25 @@ def str_get_dummies(arr, sep='|'):
     1  0  0  0
     2  1  0  1
     """
-    arr = arr.fillna('')
-    try:
-        arr = sep + arr + sep
-    except TypeError:
-        arr = sep + arr.astype(str) + sep
+    from pandas.core.reshape.reshape import get_dummies
+    from pandas import Series
+
+    if len(arr) == 0:
```

Comment:
is this case not handled by get_dummies already?

Reply:
`np.concatenate` fails with:

> stacked = Series(np.concatenate(arr_split))
> ValueError: need at least one array to concatenate

Reply:
Is there a nicer way to detect an empty input and output empty results? Should this be handled by `pandas.core.strings.StringMethods.get_dummies` instead?

Reply:
@jreback I have replaced `if len(arr) == 0: return empty` with `Series(np.concatenate(arr_split)) if len(arr) > 0 else Series()` which side steps `np.concatenate` when the input is empty. Is this satisfactory?